app/models/miq_request_workflow.rb

```diff
@@ -1075,11 +1075,8 @@ def allowed_storages(_options = {})
 
     rails_logger('allowed_storages', 0)
     st = Time.now
-    MiqPreloader.preload(hosts, :storages => {}, :host_storages => :storage)
 
-    storages = hosts.each_with_object({}) do |host, hash|
-      host.writable_accessible_storages.each { |s| hash[s.id] = s }
-    end.values
+    storages = Storage.where(:id => HostStorage.writable_accessible.where(:host => hosts).pluck(:storage_id))
```

Comment:
Great catch, what about a `join` so we don't have to bring back all of the host_storage_ids in a query then a second query for the `Storage.where` ?
`Storage.joins(:host_storages).where(:host_storages => HostStorage.writable_accessible.where(:host => hosts))`

Reply:
good catch

```suggestion
    storages = Storage.where(:id => HostStorage.writable_accessible.where(:host => hosts).select(:storage_id))
```

Reply:
Yeah good call switching to select also yields only one query